src/ReportPortal.Client/Abstractions/Resources/IAsyncLaunchResource.cs

```diff
@@ -0,0 +1,47 @@
+﻿using ReportPortal.Client.Abstractions.Requests;
+using ReportPortal.Client.Abstractions.Responses;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace ReportPortal.Client.Abstractions.Resources
+{
+    /// <summary>
+    /// Asynchronously interacts with launches.
+    /// </summary>
+    public interface IAsyncLaunchResource
+    {
+        /// <summary>
+        /// Asynchronously finishes specified launch.
+        /// </summary>
+        /// <param name="uuid">UUID of specified launch.</param>
+        /// <param name="request">Information about representation of launch to finish.</param>
+        /// <returns>A message from service.</returns>
+        Task<LaunchFinishedResponse> FinishAsync(string uuid, FinishLaunchRequest request);
```

Comment:
Is it really needed for us to have support two method: with and w/o cancellation token? Maybe better just to follow Microsoft async pattern and has only method with cancellation token? For v1 we support both version for backward campatibility.

Reply:
I even don't mind to leave methods with cancellation tokens only, even for v1, event if it is breaking change. Let's do it in separate PR.